x-pack/plugins/runtime_fields/public/components/runtime_field_form/runtime_field_form.tsx

```diff
@@ -92,10 +185,31 @@ const RuntimeFieldFormComp = ({ defaultValue, onChange, links }: Props) => {
         </EuiFlexItem>
       </EuiFlexGroup>
 
+      {existingConcreteFields.includes(name) && (
+        <>
+          <EuiSpacer />
+          <EuiCallOut
+            title={i18n.translate('xpack.runtimeFields.form.fieldShadowingCalloutTitle', {
+              defaultMessage: 'Field shadowing',
+            })}
+            iconType="pin"
+            size="s"
+            data-test-subj="shadowingFieldCallout"
+          >
+            <div>
+              {i18n.translate('xpack.runtimeFields.form.fieldShadowingCalloutDescription', {
+                defaultMessage:
+                  'This field has the same name as one of your concrete fields. This means that its value will override your mapped field.',
+              })}
+            </div>
+          </EuiCallOut>
+        </>
+      )}
+
       <EuiSpacer size="l" />
 
       {/* Script */}
-      <UseField<string> path="script">
+      <UseField<string> path="script.source">
```

Comment:
This is a comment about `links.painlessSyntax` on [line 229](https://github.com/elastic/kibana/pull/84184/files#diff-981b3ed36260a0531440cc998bf3495790e24795caf10c24564e061a734075b5R229), which just goes to the main page for the Painless guide. I don't think that information is particularly useful in this context. Instead, we could link to the section for [Mapping a runtime field](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-mapping-fields), which provides users information about writing a Painless script in the context of a runtime field. 

I believe the resulting URL would be: https://www.elastic.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-mapping-fields

Reply:
This is great idea! I will change the link 👍